airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -310,12 +338,78 @@ public void testRecordsProducedDuringAndAfterSync() throws Exception {
+    final List<AirbyteMessage> dataFromFirstBatch = AutoCloseableIterators
+        .toListAndClose(firstBatchIterator);
+    final Set<AirbyteRecordMessage> recordsFromFirstBatch = extractRecordMessages(
+        dataFromFirstBatch);
+    final List<AirbyteStateMessage> stateAfterFirstBatch = extractStateMessages(dataFromFirstBatch);
+    assertEquals(1, stateAfterFirstBatch.size());
+    assertNotNull(stateAfterFirstBatch.get(0).getData());
+    assertExpectedStateMessages(stateAfterFirstBatch);
+
+    assertEquals((MODEL_RECORDS.size()), recordsFromFirstBatch.size());
+    assertExpectedRecords(new HashSet<>(MODEL_RECORDS), recordsFromFirstBatch);
+
+    final JsonNode state = stateAfterFirstBatch.get(0).getData();
+
+    final ConfiguredAirbyteCatalog newTables = CatalogHelpers
+        .toDefaultConfiguredCatalog(new AirbyteCatalog().withStreams(List.of(
+            CatalogHelpers.createAirbyteStream(
+                    MODELS_STREAM_NAME + "_random",
+                    MODELS_SCHEMA + "_random",
+                    Field.of(COL_ID + "_random", JsonSchemaType.NUMBER),
+                    Field.of(COL_MAKE_ID + "_random", JsonSchemaType.NUMBER),
+                    Field.of(COL_MODEL + "_random", JsonSchemaType.STRING))
+                .withSupportedSyncModes(Lists.newArrayList(SyncMode.FULL_REFRESH, SyncMode.INCREMENTAL))
+                .withSourceDefinedPrimaryKey(List.of(List.of(COL_ID + "_random"))))));
+
+    newTables.getStreams().forEach(s -> s.setSyncMode(SyncMode.INCREMENTAL));
+    final List<ConfiguredAirbyteStream> combinedStreams = new ArrayList<>();
+    combinedStreams.addAll(CONFIGURED_CATALOG.getStreams());
+    combinedStreams.addAll(newTables.getStreams());
+
+    final ConfiguredAirbyteCatalog updatedCatalog = new ConfiguredAirbyteCatalog().withStreams(combinedStreams);
+
+    final AutoCloseableIterator<AirbyteMessage> secondBatchIterator = getCustomSource(newTables.getStreams())
+        .read(getConfig(), updatedCatalog, state);
+    final List<AirbyteMessage> dataFromSecondBatch = AutoCloseableIterators
+        .toListAndClose(secondBatchIterator);
+
+    final List<AirbyteStateMessage> stateAfterSecondBatch = extractStateMessages(dataFromSecondBatch);
+    final Set<AirbyteRecordMessage> recordsFromSecondBatch = extractRecordMessages(
+        dataFromSecondBatch);
+    assertEquals(1, stateAfterSecondBatch.size());
+    assertNotNull(stateAfterSecondBatch.get(0).getData());
+    assertExpectedStateMessages(stateAfterSecondBatch);
+
+    assertEquals((RANDOM_TABLE_RECORDS.size()), recordsFromSecondBatch.size());
+    assertExpectedRecords(new HashSet<>(RANDOM_TABLE_RECORDS), recordsFromSecondBatch,
+        recordsFromSecondBatch.stream().map(AirbyteRecordMessage::getStream).collect(
+            Collectors.toSet()), Sets
+            .newHashSet(MODELS_STREAM_NAME + "_random"),
+        MODELS_SCHEMA + "_random");
+  }
+
+
+  protected static final List<JsonNode> RANDOM_TABLE_RECORDS = ImmutableList.of(
```

Comment:
where are these records actually used? afaict, the tests only ever reference the size of this list, rather than its contents

also nitpick: declare this at top of class rather than at the bottom?

Reply:
It's being used in assertion 
```
    assertExpectedRecords(new HashSet<>(RANDOM_TABLE_RECORDS), recordsFromSecondBatch,
        recordsFromSecondBatch.stream().map(AirbyteRecordMessage::getStream).collect(
            Collectors.toSet()), Sets
            .newHashSet(MODELS_STREAM_NAME + "_random"),
        MODELS_SCHEMA + "_random")
```